cabal-install/src/Distribution/Client/TargetSelector.hs

```diff
@@ -815,6 +816,14 @@ reportTargetSelectorProblems verbosity problems = do
          ++ "packages in your project and all other build configuration. "
          ++ "See the Cabal user guide for full details."
 
+    case [ t | TargetSelectorNoScript t <- problems ] of
+      []  -> return ()
+      target:_ ->
+        die' verbosity $
+            "The script '" ++ showTargetString target ++ "' does not exist, "
+         ++ "and only script targets may contain whitespace characters or end "
+         ++ "with ':'"
```

Comment:
what about a test with this corner case (spaces are already being tested)? it is easy to lose that corner case

Reply:
Spaces are already being tested. I can't easily include a test with a script that ends with ':' because files that end in ':' are not valid on windows, so putting one in the git repo is a no-go.

Reply:
I can probably check the OS is the cabal.test.hs and move it to end with ':' if the os is linux.

Reply:
there is no a existing way to run tests only in a specific os (or equivalently ignore it in a specific os)?